application/src/main/java/org/togetherjava/tjbot/commands/moderation/attachment/AttachmentListener.java

```diff
@@ -0,0 +1,118 @@
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
+import net.dv8tion.jda.api.requests.RestAction;
+import org.jetbrains.annotations.NotNull;
+import org.jooq.tools.StringUtils;
+import org.togetherjava.tjbot.commands.MessageReceiverAdapter;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.moderation.ModAuditLogWriter;
+
+import java.awt.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Listener that receives all sent messages in every channel and checks if the message has any
+ * blacklisted file attached.
+ * <p>
+ * If there was a blacklisted file attached, delete the message and send the user a dm, telling what
+ * they did wrong and also inform the mods.
+ */
+public class AttachmentListener extends MessageReceiverAdapter {
+    private final Config config;
+    private final ModAuditLogWriter modAuditLogWriter;
+
+    /**
+     * Creates a AttachmentListener to receive all message sent in any channel.
+     *
+     * @param config to find the blacklisted media attachments
+     * @param modAuditLogWriter to inform the mods about the suspicious attachment
+     */
+    public AttachmentListener(@NotNull Config config,
+            @NotNull ModAuditLogWriter modAuditLogWriter) {
+        super(Pattern.compile(".*"));
+        this.config = config;
+        this.modAuditLogWriter = modAuditLogWriter;
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        if (event.getAuthor().isBot() || event.isWebhookMessage()) {
+            return;
+        }
+        if (attachmentContainsBlacklistedFileExtension(event.getMessage())) {
+            deleteMessageAndDmUser(event);
+            warnMods(event);
+        }
+    }
+
+    private void deleteMessageAndDmUser(@NotNull MessageReceivedEvent event) {
+        event.getMessage().delete().flatMap(any -> dmUser(event)).queue();
+    }
+
+    private RestAction<Message> dmUser(@NotNull MessageReceivedEvent event) {
+        Message dmMessage = createDmMessage(event.getMessage());
+        return event.getJDA()
+            .openPrivateChannelById(event.getAuthor().getIdLong())
+            .flatMap(channel -> channel.sendMessage(dmMessage));
+    }
+
+    private Message createDmMessage(Message originalMessage) {
+        String originalMessageContent = originalMessage.getContentDisplay();
+        String blacklistedAttachments = getBlacklistedAttachmentsFromMessage(originalMessage);
+        String dmMessageContent =
+                "Hey there, you posted a message with a blacklisted file attachment: %s. Following file extension are blacklisted: %s ."
+                    .formatted(blacklistedAttachments, config.getBlacklistedFileExtensions());
+        // No embedded needed if there was no message from the user
+        if (StringUtils.isEmpty(originalMessageContent)) {
```

Comment:
What is `StringUtils`? Do we need that? Why not just `originalMessageContent.isEmpty()`?